keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -397,6 +385,12 @@ open class SqlResourceRepository(
         ?: throw IllegalStateException("Resource with id $id not found. Retrying.")
     }
 
+  private fun applicationForId(id: String): Select<Record1<String>> =
```

Comment:
Just for my education, is there an advantage to returning the JOOQ select record here instead of doing the `fetch` and returning the value?

Reply:
yes, because then you can use it to compose a larger query. If you return the value from the function you'll always be executing this as a separate query.